src/app/map/shared/layers/layer-wmts.ts

```diff
@@ -0,0 +1,56 @@
+import { Layer, LayerOptions} from './layer';
+import 'rxjs/add/operator/toPromise';
+
+export interface WMTSLayerOptions extends LayerOptions {
+  source: olx.source.WMTSOptions;
+  view?: olx.layer.TileOptions;
+  optionsFromCapabilities?: boolean;
+}
+
+export class WMTSLayer extends Layer {
+
+  olLayer: ol.layer.Tile;
+  getCapabilitiesResponse: any;
+
+  createOlLayer(options: WMTSLayerOptions,
+                getCapabilities: ol.format.WMTSCapabilities): ol.layer.Tile {
+
+    if ( getCapabilities ) {
+
+      const olLayerOptionsGetCapabilities = ol.source.WMTS.optionsFromCapabilities(getCapabilities,
+                                                                                    options.source);
+
+      const olLayerOptions = Object.assign(options.view || {}, {
+        source: new ol.source.WMTS(olLayerOptionsGetCapabilities)
+      });
+      return new ol.layer.Tile(olLayerOptions);
+
+    } else {
+
```

Comment:
I think this would be a cool utils method because we might need it somewhere else later!